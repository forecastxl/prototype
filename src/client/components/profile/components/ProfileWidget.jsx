import PropTypes from 'prop-types'
import React from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { IconMenu, MenuItem, Avatar } from 'material-ui'
import { HardwareKeyboardArrowDown, ActionSettings, ActionExitToApp } from 'material-ui/svg-icons'
import styled from 'styled-components'
import { colors } from '../../../styles'
import { fetchProfile } from '../actions'
import { logout } from '../../../data/session/actions'
import { getProfile } from '../selectors'

const mapStateToProps = state => ({ profile: getProfile(state) })
const mapStateToDispatch = dispatch => ({
  fetchProfile: () => dispatch(fetchProfile()),
  logout: () => dispatch(logout())
})

const ProfileName = styled.span`
  color: ${colors.white};
  font-size: 0.8rem;
  font-weight: 200;
  margin-left: 15px;
`

const ProfileWidgetContent = styled.div`
  & * {
    vertical-align: middle;
  }
  &:hover {
    cursor: pointer;
  }
`

export class DumbProfileWidget extends React.Component {
  componentWillMount() {
    this.props.fetchProfile()
  }

  render() {
    const { profile } = this.props
    return (
      <IconMenu
        anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
        targetOrigin={{ horizontal: 'right', vertical: 'top' }}
        iconButtonElement={
          <div>
            {/* Note: must be wrapped in a div, otherwise click events are not properly attached */}
            <ProfileWidgetContent>
              <Avatar size={35} src="http://i.pravatar.cc/150?u=a042581f4e290267014f" />
              <ProfileName>{profile.name}</ProfileName>
              <HardwareKeyboardArrowDown
                color={colors.white}
                style={{ marginLeft: '2px', marginTop: '2px' }}
              />
            </ProfileWidgetContent>
          </div>
        }
      >
        <MenuItem
          leftIcon={<ActionSettings />}
          primaryText="Instellingen"
          onTouchTap={() => {
            this.props.history.push('/settings')
          }}
        />
        <MenuItem
          leftIcon={<ActionExitToApp />}
          primaryText="Uitloggen"
          onTouchTap={() => {
            this.props.logout()
            this.props.history.push('/login')
          }}
        />
      </IconMenu>
    )
  }
}

DumbProfileWidget.propTypes = {
  history: PropTypes.object.isRequired,
  profile: PropTypes.object.isRequired,
  fetchProfile: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapStateToDispatch)(withRouter(DumbProfileWidget))
