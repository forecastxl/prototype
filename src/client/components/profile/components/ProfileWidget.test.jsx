import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { DumbProfileWidget } from './ProfileWidget'

describe('<DumbProfileWidget />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <DumbProfileWidget
        history={{}}
        profile={{
          name: 'John Doe'
        }}
        fetchProfile={() => {}}
        logout={() => {}}
      />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
