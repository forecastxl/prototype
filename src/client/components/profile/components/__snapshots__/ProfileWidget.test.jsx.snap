// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DumbProfileWidget /> renders correctly 1`] = `
<IconMenu
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "bottom",
    }
  }
  animated={true}
  iconButtonElement={
    <div>
      <styled.div>
        <Avatar
          size={35}
          src="http://i.pravatar.cc/150?u=a042581f4e290267014f"
        />
        <styled.span>
          John Doe
        </styled.span>
        <HardwareKeyboardArrowDown
          color="#FFFFFF"
          style={
            Object {
              "marginLeft": "2px",
              "marginTop": "2px",
            }
          }
        />
      </styled.div>
    </div>
  }
  multiple={false}
  onItemTouchTap={[Function]}
  onKeyboardFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onRequestChange={[Function]}
  onTouchTap={[Function]}
  open={null}
  targetOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  touchTapCloseDelay={200}
  useLayerForClickAway={false}
>
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    leftIcon={<ActionSettings />}
    onTouchTap={[Function]}
    primaryText="Instellingen"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  />
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    leftIcon={<ActionExitToApp />}
    onTouchTap={[Function]}
    primaryText="Uitloggen"
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  />
</IconMenu>
`;
